src/instruments/src/RMP/AcceleratedKnob.mjs

```diff
@@ -0,0 +1,34 @@
+/**
+ * A generic rotary encoder knob which handles acceleration.
+ * If the knob is turned quickly the speed given to the callback increases.
+ * When the knob is not used for longer than the timeout, the speed goes back to 1.
+ *
+ * E.g. this can be used to increase the delta heading for a heading knob
+ * as a user turns faster as opposed to always changing the heading by 1 degree.
+ */
+export class AcceleratedKnob {
+    constructor(timeout = 300, increment = 0.20) {
+        this._currentSpeed = 0;
+        this._timeout = timeout;
+        this._increment = increment;
+        this._previousTimestamp = 0;
+    }
+
+    increase() {
+        if (this._currentSpeed < 1 || (Date.now() - this._previousTimestamp) > this._timeout) this._currentSpeed = 1;
```

Comment:
So generally in a Javascript world Date.now() is fine. However, because we're in a Sim here in future, imagine a possible replay mode in future that replays events, but the user chooses 0.5x as the replay speed, or 2x. While that is a hypothetical, I'm generally hesitant of referencing the out-of-sim real-world time from inside our Javascript, especially as the sim already passes us a generic deltaTime (which we have access for example in React with the appropriate hook).

I don't have a concrete suggestion here, especially as it would require a significant rewrite, but I would just like to point out that referencing real-world time should not be done as lightly in the sim as on a website.

Reply:
More research is required to see how this would perform in edge cases (time acceleration, playback, etc.)
I think this is good enough for now.